### DISCUSSION TITLE
Are there any plans to integrate CDK instead of cloud formation for copilot?

### DISCUSSION THREAD

**Discussion by DennisSSDev** [aws/copilot-cli#2472] [General]
It seems that [CDK](https://github.com/aws/aws-cdk) is becoming more and more popular for defining AWS infra programmatically. I was wondering, are there any plans to give us the option to write our "addons" and infra using CDK, instead of writing cloud formation-like yamls?

**Comment by efekarakus** [comment]
Hi @DennisSSDev !

Yesss this is definitely an integration that we are interested in! It's not listed in the roadmap yet until we have an outline defined, but it'd be great if we can use this discussion post to gather use cases.

> to write our "addons" and infra using CDK

Can you tell me more about what you mean by: "and infra"? Are there other pieces of infrastructure besides addons you'd like to define using the CDK?

**Comment by afgallo** [comment]
I can definitely see a use case for creating addons with CDK rather than pure Cloudformation. @efekarakus how can we vote on this to make sure you guys can prioritise in the future?

**Comment by efekarakus** [comment]
Hi folks! We just posted https://github.com/aws/copilot-cli/issues/4208 as a proposal for extending Copilot with the CDK.

The proposal should enable you to write L2 constructs with the CDK under the `overrides/` directory. We would love to hear your feedback ❤️
